concepts/type-checking/about.md

```diff
@@ -0,0 +1,144 @@
+# About
+
+Knowning what type an object has is often very important for code to run smoothly and without errors.
+
+Javascript has several ways to check the type of an object.
+
+```exercism/note
+Javascript's type checking mechanisms are always soomewhat unreliable.
+
+For true safety with types, you should probably use TypeScript, a language that builds on JavaScript, but with the type syntax of a static-typed language.
+```
+
+## The `typeof` operator
+
+The `typeof` operator returns the type of its input.
+The output is restricted to one of the [primitive data types][primitives], `"function"` or `"object"`.
+
+```javascript
+typeof undefined;
+// => "undefined"
+
+typeof true;
+// => "boolean"
+
+typeof 42;
+// => "number"
+
+typeof 'Hello, World!';
+// => "string"
+
+typeof function () {
+  return 'Hello, World';
+};
+// => "function"
+
+typeof [1, 2, 3, 4];
+// => "object"
+
+typeof { city: 'Stockholm', country: 'Sweden' };
+// => "object"
+```
+
+The one exception to this is that `typeof null` returns `"object"` for [historical reasons][typeof null is object].
+
+## The `instanceof` operator
+
+For checking the type of an object, you can use the `instanceof` operator.
+It returns a boolean depending on whether the second operand is included in the first operands' [prototype chain][prototype chain].
+To clarify, `instanceof` will return whether the first operand is an instance of second operand or one of its child classes.
+`instanceof` only works for compound data types, such as arrays and objects.
```

Comment:
What are compound data types? What are they not? I don't know, but if you tell me I can probably rewrite this sentence ot be clearer.